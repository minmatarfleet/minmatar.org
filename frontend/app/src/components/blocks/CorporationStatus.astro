---
import { i18n } from '@helpers/i18n'
const { t, translatePath } = i18n(Astro.url)

import type { Alert, CorporationObject } from '@dtypes/layout_components'

interface Props {
    corporation:            CorporationObject;
    is_user_corporation?:   boolean;
    alert?:                 Alert | false;
    disable_applications?:  boolean;
}

const {
    corporation,
    is_user_corporation = false,
    alert = false,
    disable_applications = false
} = Astro.props

const primary_character = Astro.cookies.has('primary_pilot') ? JSON.parse(Astro.cookies.get('primary_pilot')?.value as string) : null
const primary_character_id = primary_character ? parseInt(primary_character.character_id) : null

import { query_string } from '@helpers/string';
const CORPORATION_REQUEST_STATUS_PARTIAL_URL = translatePath('/partials/corporation_request_status_item/')

import ButtonStack from '@components/blocks/ButtonStack.astro';
import Button from '@components/blocks/Button.astro';
import ButtonAuth from '@components/blocks/ButtonAuth.astro';
import Tag from '@components/blocks/Tag.astro';
import ButtonAddPrimary from '@components/blocks/ButtonAddPrimary.astro';
import ButtonCeoToken from '@components/blocks/ButtonCeoToken.astro';
---

<div
    id={`corporation-status-${corporation.corporation_id}`}
    x-data={`{
        init() {
            ${alert !== false ? `show_alert_dialog(${JSON.stringify(alert)})` : ''}
        }
    }`}
>
    {(corporation.active && is_user_corporation) ?
        <div class="[ max-w-fit ]">
            <Tag color='green'
                data-tippy-content={t('joined_corporation_tooltip')}
                x-init="tippy($el, tippy_options)"
                text={t('joined')}
            />
        </div>
        :
        (corporation?.status != null && !corporation.active) ?
            <ButtonStack>
                <ButtonCeoToken
                    corporation={corporation}
                    size='sm'
                    dialog_title={t('add_ceo_token_dialog_title')}
                >
                    {t('add_ceo_token')}
                </ButtonCeoToken>
            </ButtonStack>
        :
        !disable_applications &&
            <ButtonStack>
                {(corporation?.status != null && !corporation.active) ?
                    <ButtonCeoToken
                        corporation={corporation}
                        size='sm'
                        dialog_title={t('add_ceo_token_dialog_title')}
                    >
                        {t('add_ceo_token')}
                    </ButtonCeoToken>
                :
                (corporation?.status == 'available' && !primary_character_id) ?
                    <ButtonAddPrimary
                        size='sm'
                        color='green'
                        dialog_title={t('corporation_join_request_dialog_title')}
                    >
                        {t('enlist_today')}
                    </ButtonAddPrimary>
                :
                (corporation?.status == 'available' && primary_character_id) ?
                    <Button
                        type="button"
                        size='sm'
                        color='green'
                        x-bind:disabled="disabled"
                        x-data={`{
                            disabled: false,
                            show_join_request_dialog() {
                                this.disabled = true
                                
                                show_confirm_dialog({
                                    title: '${t('corporation_join_request_dialog_title')}',
                                    partial: '${translatePath('/partials/dialog_apply_corporation/')}?${query_string({
                                        corporation_id: corporation.corporation_id,
                                        corporation_name: corporation.corporation_name,
                                        alliance_id: corporation.alliance_id,
                                        alliance_name: corporation.alliance_name,
                                        corporation_type: corporation.corporation_type,
                                        status: corporation.status,
                                        message: t('corporation_join_request_dialog_text')
                                    })}',
                                    return_on_accept: ${corporation.corporation_id},
                                    hx: {
                                        method: 'post',
                                        url: '${CORPORATION_REQUEST_STATUS_PARTIAL_URL}?${query_string({
                                            corporation: JSON.stringify(corporation),
                                        })}',
                                        target: '${`#corporation-status-${corporation.corporation_id}`}',
                                        swap: 'outerHTML transition:true'
                                    }
                                }).then( (corporation_id) => { this.disabled = !!corporation_id } )
                            }
                        }`}
                        x-on:click="show_join_request_dialog()"
                        type="button"
                    >
                        {t('enlist_today')}
                    </Button>
                :
                corporation?.status == 'unauth' ?
                    <ButtonAuth
                        size='sm'
                        color='green'
                        dialog_title={t('corporation_join_request_dialog_title')}
                    >
                        {t('enlist_today')}
                    </ButtonAuth>
                :
                corporation?.status === 'accepted' ?
                    <Tag
                        data-tippy-content={t('application_accepted_tooltip')}
                        x-init="tippy($el, tippy_options)"
                        color='green'
                        text={t('accepted')}
                    />
                :
                corporation?.status === 'rejected' ?
                    <Tag
                        data-tippy-content={t('application_denied_tooltip')}
                        x-init="tippy($el, tippy_options)"
                        text={t('denied')}
                    />
                :
                corporation?.status == 'pending' ?
                    <Tag
                        data-tippy-content={t('pending_application_tooltip')}
                        x-init="tippy($el, tippy_options)"
                        color='alliance-blue' text={t('pending')}
                    />
                :
                corporation?.status == 'error' &&
                    <Tag
                        data-tippy-content={t('error_reload')}
                        x-init="tippy($el, tippy_options)"
                        text={t('error_status_corp')}
                    />
                }
            </ButtonStack>
    }
</div>