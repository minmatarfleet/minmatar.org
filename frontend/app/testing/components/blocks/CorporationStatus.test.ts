import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import CorporationStatus from "@components/blocks/CorporationStatus.astro";

test("CorporationStatus defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(CorporationStatus, {
    props: {
      corporation: {
        corporation_id: 87,
      },
    },
    slots: {
      default: "slotted chick",
    },
  });

  // expect(result).toMatchSnapshot();
});
