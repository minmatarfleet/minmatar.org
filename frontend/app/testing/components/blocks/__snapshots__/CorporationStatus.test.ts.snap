// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CorporationStatus defaults 1`] = `
"<div id="corporation-status-87" x-data="{
        init() {
            
        }
    }"> <div class="flex-block [ button-stack ] gap---space-2xs" data-mod style="--flex_block_gap: var(--space-2xs);">    </div>   </div>"
`;
